### ISSUE TITLE
Vue Demi dependency broken

### ISSUE THREAD

**Issue by ESP-Marc** [posva/pinia#2403]
### Reproduction

Can't reproduce with CDN assets

### Steps to reproduce the bug

Install Pinia on a Vue 2.7 project and run it

### Expected behavior

Build/serve without error

### Actual behavior

Builds with an error relating to `vueuse/vue-demi`

```
The requested module 'vue-demi' does not provide an export named 'hasInjectionContext'
```

### Additional information

I have raised an issue with Vue Demi Lib here: https://github.com/vueuse/vue-demi/issues/244

I just wanted to make the Pinia community here aware of this bizarre issue that has just cropped up in in the last few days where `vue-demi@v1.14.6`  is not working for my team with Pinia on Vue 2.7 and when we try force `vue-demi@v1.14.5` that we know has been working fine for the past few months since May where `Pinia@2.1.1` has consumed that particular version of Demi `.5` up-until this week just fine, it has suddenly started to fail with same issue as `.6` release this week.

As I have outlined in the linked issue in the other lib, I believe `vue-demi@v1.14.5` has been overwritten by accident as I have two totally different copies of the same version of this dependency?

**Comment by posva** [comment]
Maybe you need to use pnpm dedupe after updating Vue-Demi but in my tests it works

**Comment by ESP-Marc** [comment]
@posva Fraid not, no dupes reported at all

Let me reiterate:
11 Days ago `vue-demi` released version `0.14.6` < This version is broken for us because its Vue version detection is not working, it loads Vue 3 `/lib/v3/` compatibility files in our Vue 2.7 project instead of from `/lib/v2.7/`.

So to fix this, I tried forcing Pinia to consume `0.14.5` that I know has been working since the middle of May this year by using `resoltions`.

```
  "resolutions": {
    "vue-demi": "0.14.5"
  }
```

However much to my surprise, this did not work when I confirmed my node_modules directory had `0.14.5` in it for deffinate, I was puzzled when I got the same error in `0.14.5` as I do with `0.14.6`.
When I investigated this, I found that my old copy of `0.14.5` from another Pina project is totally different from the `0.14.5` I downloaded this week in a fresh clone.

To be absolutely clear, `0.14.5` is not the same as it was a few months ago when it was originally npm published:

![same-version-different-dates](https://github.com/vuejs/pinia/assets/116962814/60456da1-c884-48ef-85cf-e36a02377628)

The `package.json` in the old copy (left) vs recent copy (right) both sate in the file the same version:

```
  "name": "vue-demi",
  "version": "0.14.5",
```
